### ISSUE TITLE
Improve TK death detection by using FSM

### ISSUE THREAD

**Issue by royitaqi** [royitaqi/HollowKnight.GodhomeWinLossTracker#2]
Chat:
https://discord.com/channels/879125729936298015/879130756146954240/988563017425903637

By Korz:
```
PlayMakerFSM fsm = GameObject.Find("Knight").transform.Find("Hero Death").gameObject.LocateMyFSM("Hero Death Anim");
                fsm.GetState("Anim Start").AddMethod(()
                    =>
                {
                    Log("It's a dream :OOO");
                    PlayerData.instance.SetInt("geo", 420);
                });
```
> When I was killed by Failed Champ:
```
[INFO]:[DebugMod] - [DREAM GATE] File dreamgate.dat not found!
[INFO]:[LoreMaster] - It's a dream :OOO
```

**Comment by royitaqi** [comment]
> Roy:
> Can a hook like this be added to the modding api, so that noobs like me can hook the death animation (without understanding FSM)?
> Or, is the code example simply addable to a mod's initialization code without too much setup? 


> Someone:
> Probably not to the api since modifying fsms would require more stuff in preexisting hooks, and it would be modifying fsms from what people see in fsmviewer
> If it did modify fsms, it should probably be done in the game’s asset files so fsmviewer would load them the same way

**Comment by royitaqi** [comment]
> Clazex:
> we already have On.PlayMakerFSM.Start

**Comment by royitaqi** [comment]
> Korz
> You could just implement the example and replace my method with your own one (or use the solution with itemchanger). Take Note that my logic will most likely only work once you entered gameplay scene because I assume the knight object is not active in the menu.   Therefore you need to hook on something that happens in game play (or just check the current scene)

**Comment by royitaqi** [comment]
Implemented in 64b7d7ef00edc71ac9b726611b5bcb7f6d81135a and branch issue-2.
